### ISSUE TITLE
@cascade disables value edge facet fetching in PR #4267

### ISSUE THREAD

**Issue by robsws** [dgraph-io/dgraph#4310]
I am building Dgraph from source using the animesh2049/issue_4081 branch currently up for review in PR #4267.

Using the following data:
```
{
   set {
     _:alice <Name> "Alice Tester" (since=2019-01-01) .
   }
}
```

I am able to fetch the "since" facet back without the `@cascade `directive, but not with it. To demonstrate, running this query:
```
{
  query(func:eq(Name, "Alice Tester")) {
    Name @facets(since)
  }
}
```
gives this result:
```
    "query": [
      {
        "Name|since": {
          "0": "2019-01-01T00:00:00Z"
        },
        "Name": [
          "Alice Tester"
        ]
      }
    ]
```

And this query, with` @cascade`:
```
{
  query(func:eq(Name, "Alice Tester")) @cascade {
    Name @facets(since)
  }
}
```
gives this result:
```
    "query": [
      {
        "Name": [
          "Alice Tester"
        ]
      }
    ]
```

As far as I know, `@cascade` should not be affecting whether or not facets are returned for a value edge. As a further note, this problem does not seem to occur for uid edges, only value edges.

**Comment by shekarm** [comment]
Hello,

Thank you for your question. Let me find someone within our team to address it.

**Comment by MichelDiz** [comment]
Hi guys,

I'm not 100% about this behavior. For me cascade is related to nodes only. Facets aren't first-class-citizens do the majority of the functions are' designed to work with them.

I'll let someone with the deepest knowledge in the cascade design work out what to do better or not.

Cheers.

**Comment by MichelDiz** [comment]
BTW, I believe this is the best work around you can have

```
{
  T as var(func: type(User)) @cascade {
    Name
    age
  }
  
  query(func: uid(T)){
    Name @facets(since)
    age
  }

}
```

This way you can isolate the cascade from the facet.

Actually I think this is the solution. If so, case closed.

**Comment by sleto-it** [comment]
Hi all,
This ticket was fixed by PR #4530, which has now been merged

The fix will be delivered in v1.2

Please comment if more help is needed, and we will reopen

Many thanks,

**Comment by robsws** [comment]
Thanks for the fix here. I can confirm it looks good in 1.2.
